### ISSUE TITLE
Db.query not work

### ISSUE THREAD

**Issue by sergq78** [hgourvest/node-firebird#202]
Hello.
Last version. 
Db.query not return anything. In to function(err, result).
This part of code is not run
But i see query to db in ibexpert trace.

**Comment by maratth** [comment]
You can provide your query and Firebird version ?

**Comment by sergq78** [comment]
main function

async function loadAllUsers () {
	console.log('asdasdasdasdasd0');
	var z = await WorkDb('select * from site_users where site_users_type_id = 1');
	console.log('after await');
	redisClient.select(config.redisStore.db);
	for (var i = 0; i < z.length; i++) {
		var o;// = z[i];
			{
			o = new Object();
			o.userName = z[i].U; // 
			o.pass = z[i].PASS.trim();
			o.Utype = 1; 
			redisClient.set(o.userName,JSON.stringify(o));//,redis.print
		}

	}
	console.log('users loaded - '+z.length);
	
	
}

async function WorkDb (s) {
		
		return new Promise (function (resolve,reject) {
			pool.get(function(err, db) {
					console.log('db done');
					if (err) {
						console.log('db err');
						console.log('pool.get error 31');
						console.log(err);
						db.detach();	
						res.status(503).end();
						return;
								
					}
				
				db.query(s, function(err, result) { //	
				console.log('db done1');
						if (err) {
							//throw err;
							reject(err);
						}
						
						db.detach();
						resolve(result);
				});
			
			});
			
		})
		
		
};


in WorkDb i see db done. but not see db done1

**Comment by sergq78** [comment]
some times i see "db done1". some times not.

**Comment by maratth** [comment]
Fb 2.5 or 3 ?

**Comment by sergq78** [comment]
2.5

**Comment by maratth** [comment]
It's work with 0.8.10 version ?

I can't reproduce with this sample test (this test work fine) :
```js
pool.get(function (err, db) {
    assert.ok(!err, err);

    db.query('SELECT FIRST 10 * FROM RDB$CHARACTER_SETS', [], (err, data) => {
        assert.ok(!err, err);
        assert.ok(data.length === 10);
        
        db.detach();
        done();
    });
});
```

**Comment by sergq78** [comment]
my code work fine on 0.8.9

**Comment by sergq78** [comment]
found )
if i run in debugger step by step Database.prototype.execute  with run step by step transaction.execute - all fine. if run without debugger - Database.prototype.execute not return callback

**Comment by mcvendrell** [comment]
Oh, I'm getting the same problem, but downgrading to 0.8.9 nor 0.8.10 resolve the issue :(

**Comment by jucapablanca** [comment]
Install lasted 0.9.9 version and check again, this bugs has fixed on lasted  version.

@mapopa close this issue
